syntax = "proto3";

package second.protos;
import "rslo/protos/preprocess.proto";
import "rslo/protos/sampler.proto";

message InputReader {
  uint32 batch_size = 1;
 
  message Dataset {
    string kitti_info_path = 1;
    string kitti_root_path = 2;
    string dataset_class_name = 3; // support KittiDataset and NuScenesDataset
    int32 seq_length=4;
    int32 skip=5;
    bool random_skip=6;
    int32 step=7;
    
  }
  Dataset dataset = 2;
  message Preprocess {
    bool shuffle_points = 1;
    uint32 max_number_of_voxels = 2;
    repeated float groundtruth_localization_noise_std = 3;
    repeated float groundtruth_rotation_uniform_noise = 4;
    repeated float global_rotation_uniform_noise = 5;
    repeated float global_scaling_uniform_noise = 6;
    repeated float global_translate_noise_std = 7;
    bool remove_unknown_examples = 8;
    uint32 num_workers = 9;
    float anchor_area_threshold = 10;
    bool remove_points_after_sample = 11;
    float groundtruth_points_drop_percentage = 12;
    uint32 groundtruth_drop_max_keep_points = 13;
    bool remove_environment = 14;
    repeated float global_random_rotation_range_per_object = 15;
    repeated DatabasePreprocessingStep database_prep_steps = 16;
    Sampler database_sampler = 17;
    bool use_group_id = 18; // this will enable group sample and noise
    int64 min_num_of_points_in_gt = 19; // gt boxes contains less than this will be ignored.
    bool random_flip_x = 20;
    bool random_flip_y = 21;
    float sample_importance = 22;
    float rand_rotation_eps=23;
    float rand_translation_eps=24;
    float random_aug_ratio=25;
    float do_pre_transform=26;
    bool cubic_tq_map=27;
    repeated float downsample_voxel_sizes=28;
  }
  Preprocess preprocess = 3;
  uint32 max_num_epochs = 4; // deprecated
  uint32 prefetch_size = 5;  // deprecated

  float review_cycle=6;
}
